test/unit/io/sequence_file/sequence_file_format_genbank_test.cpp

```diff
@@ -86,40 +58,81 @@ ORIGIN
 )"
     };
 
-    detail::sequence_file_input_format_REMOVEME<format_genbank> format;
+    std::string illegal_alphabet_character_input
+    {
+R"(LOCUS ID1
+DEFINITION  Homo sapiens mRNA for prepro cortistatin like peptide, complete
+            cds.
+ACCESSION   ID1
+ORIGIN
+        1 ACGTTTT?TT TTTTTTTT
+//
+)"
+    };
 
-    sequence_file_input_options<dna5, false> options;
+    std::string standard_output
+    {
+R"(LOCUS       ID1                 18 bp
+ORIGIN
+        1 ACGTTTTTTT TTTTTTTT
+//
+LOCUS       ID2                 82 bp
+ORIGIN
+        1 ACGTTTTTTT TTTTTTTTTT TTTTTTTTTT TTTTTTTTTT TTTTTTTTTT TTTTTTTTTT
+       61 TTTTTTTTTT TTTTTTTTTT TT
+//
+LOCUS       ID3 lala                 7 bp
+ORIGIN
+        1 ACGTTTA
+//
+)"
+    };
 
-    std::string id;
-    dna5_vector seq;
+    std::string no_or_ill_formatted_id_input
+    {
+R"(LOCOS ID1    stuff
+ORIGIN
+        1 ACGTTTTTTT TTTTTTTT
+//
+)"
+    };
+};
+
+// ---------------------------------------------------------------------------------------------------------------------
+// parametrized tests
+// ---------------------------------------------------------------------------------------------------------------------
+
+INSTANTIATE_TYPED_TEST_CASE_P(genbank, sequence_file_read, format_genbank);
+INSTANTIATE_TYPED_TEST_CASE_P(genbank, sequence_file_write, format_genbank);
+
+// ----------------------------------------------------------------------------
+// reading
+// ----------------------------------------------------------------------------
+
+struct read : public sequence_file_read<format_genbank>
+{
+    sequence_file_input_options<dna15, false> options;
 
     void do_read_test(std::string const & input)
     {
         std::stringstream istream{input};
 
-        for (unsigned i = 0; i < 3; ++i)
-        {
-            id.clear();
-            seq.clear();
+        sequence_file_input fin{istream, format_genbank{}};
+        fin.options = options;
 
-            EXPECT_NO_THROW(( format.read(istream, options, seq, id, std::ignore) ));
-            EXPECT_EQ(id, expected_ids[i]);
-            EXPECT_EQ(seq, expected_seqs[i]);
-            EXPECT_TRUE((ranges::equal(seq, expected_seqs[i])));
-            EXPECT_TRUE((ranges::equal(id, expected_ids[i])));
+        auto it = fin.begin();
+        for (unsigned i = 0; i < 3; ++i, ++it)
+        {
+            EXPECT_EQ(get<field::ID>(*it), ids[i]);
```

Comment:
ok, but what happens with no_throw? This is not relevant anymore?

Reply:
well the reading is done while the iterator `it` is incremented. 
I could move the incrementation from inside the for loop definition `()` to inside the loop scop and wrap it in no throw again but the test will fail if the operator throws anyway AND I think the error message without the expect is much more readable... (Besides we have a lot of functions throughout the library that may throw which we did not wrap in EXPECT_NO_THROW).